api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -2190,6 +2190,27 @@ message UpsertTokenV2Request {
   }
 }
 
+// Request for GetHeadlessAuthentication.
+message GetHeadlessAuthenticationRequest {
+  // ID is the headless authentication id.
+  string id = 1;
+}
+
+// Request for UpdateHeadlessAuthenticationState.
+message UpdateHeadlessAuthenticationStateRequest {
+  // ID is the headless authentication id.
+  string id = 1;
+
+  // State is the state that the request will be updated to.
+  // MFA verification is required if State=APPROVED.
+  types.HeadlessAuthenticationState state = 2;
+
+  // MFAResponse is an mfa challenge response used to verify the user.
+  // MFA Auth Challenges can be created for a user with the
+  // authservice.GenerateAuthenticateChallenge rpc.
+  MFAAuthenticateResponse mfa_response = 3;
```

Comment:
In standard [proto style](https://protobuf.dev/programming-guides/style/#message-field-names), fields are suppose to be snake_case. Since the Buf changes, we are in an awkard in between. If you look in `types.proto`, you'll see that some of the more recent additions are mixed between styles. But new proto messages and fields would ideally be in line with proto style so the eventual removal of `gogoproto` will be less painful.